awx/main/tasks.py

```diff
@@ -342,11 +342,17 @@ def cluster_node_heartbeat(self):
             stop_local_services(['uwsgi', 'celery', 'beat', 'callback'], communicate=False)
             raise RuntimeError("Shutting down.")
     for other_inst in lost_instances:
-        if other_inst.capacity == 0:
-            continue
         try:
-            other_inst.capacity = 0
-            other_inst.save(update_fields=['capacity'])
+            # Capacity could already be 0 because:
+            #  * It's a new node and it never had a heartbeat
+            #  * It was set to 0 by another tower node running this method
+            #  * It was set to 0 by this node, but auto deprovisioning is off
+            #
+            # If auto deprovisining is on, don't bother setting the capacity to 0
+            # since we will delete the node anyway.
+            if other_inst.capacity != 0 and not settings.AWX_AUTO_DEPROVISION_INSTANCES:
+                other_inst.capacity = 0
+                other_inst.save(update_fields=['capacity'])
             logger.error("Host {} last checked in at {}, marked as lost.".format(
                 other_inst.hostname, other_inst.modified))
```

Comment:
I think we risk log flooding here, because when an instance goes down without AWX_AUTO_DEPROVISION_INSTANCES on, then we expect to be passing by it constantly in this loop by every instance in the cluster. So I think the log statement should be indented.

 Previously, I think I had considered limiting the filter for the `instance_list` queryset to exclude instances with capacity of 0. That has some appeal, but it would require excluding itself, and it just gets complicated.